### PR TITLE
Change AnalyzerCommand.cs to be testable and write tests.

### PR DESCRIPTION
**Pull request by bigfluffycookie**

### CODE REVIEW

File: Exercise/VSIX/AnalyzeCommand.cs [hunk 1]
```diff
@@ -26,45 +27,29 @@ internal sealed class AnalyzeCommand
...
+        private readonly ILogger logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AnalyzeCommand"/> class.
         /// Adds our command handlers for menu (commands must exist in the command table file)
         /// </summary>
         /// <param name="package">Owner package, not null.</param>
         /// <param name="commandService">Command service to add command to, not null.</param>
-        private AnalyzeCommand(AsyncPackage package, OleMenuCommandService commandService)
+        internal AnalyzeCommand(IMenuCommandService commandService, ILogger logger,
```
Comment (bigfluffycookie) [inline Exercise/VSIX/AnalyzeCommand.cs:42]: Changed this to make it more testable

File: Exercise/VSIX/AnalyzeCommand.cs [hunk 2]
```diff
@@ -26,45 +27,29 @@ internal sealed class AnalyzeCommand
         /// </summary>
         public static readonly Guid CommandSet = new Guid("9d169479-372d-4b21-a855-d4449ad5b742");
 
-        /// <summary>
-        /// VS Package that provides this command, not null.
-        /// </summary>
-        private readonly AsyncPackage package;
+        private readonly IAnalysisController analysisController;
```
Comment (bigfluffycookie) [inline Exercise/VSIX/AnalyzeCommand.cs:30]: Renamed this to match class name

File: Exercise/VSIX.UnitTests/AnalyzerCommandTests.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            var logger = new Mock<ILogger>();
+            var dte = new Mock<DTE2>();
+            dte.Setup(p => p.ActiveDocument).Returns((Document)null);
+            var analysisController = new Mock<IAnalysisController>();
+
+            var testSubject = new AnalyzeCommand(Mock.Of<IMenuCommandService>(), logger.Object, analysisController.Object, dte.Object);
+            testSubject.Analyze();
+
+            logger.Verify(p => p.LogWithNewLine("No file is currently active. Please open a document and try again."), Times.Once);
```
Comment (duncanp-sonar) [inline Exercise/VSIX.UnitTests/AnalyzerCommandTests.cs:57]: Minor - you could verify that the `dte.ActiveDocument` mock is called once.

**Review by duncanp-sonar — APPROVED** [review]
Nice! One minor comment about a possible additional assertion.

**Pull request merged**